src/groupeddataframe/groupeddataframe.jl

```diff
@@ -323,6 +323,20 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::Dict{T, S}, gd) where {T, S}
+    t = ntuple(length(gd.cols)) do i 
+        key[T(_names(gd)[i])]
```

Comment:
We should access columns of appropriate index (also make sure this is tested when e.g. columns 5,3, and 7 are grouping columns in the order I have given)
```suggestion
        key[T(_names(gd)[gd.cols[i]])]
```